plugin/src/sbt-test/swagger/compile/test

```diff
@@ -1,17 +1,55 @@
-# Fails while compiled for the first time, this should be addressed
+# Project is clean, nothing is here
+
+- $ exists app/controllers/echo.api.yaml.scala
+- $ exists target/scala-2.11/routes/main/controllers_base/echo.api.yaml.scala
+- $ exists target/scala-2.11/routes/main/model/echo.api.yaml.scala
+- $ exists target/scala-2.11/routes/main/validators/echo.api.yaml.scala
+- $ exists target/scala-2.11/routes/test/tests/echo.api.yaml.scala
+- $ exists target/scala-2.11/routes/test/generators/echo.api.yaml.scala
+
+# Fails while compiled for the first time because of missing controllers
 -> compile
 
-# Successfully compiles second time
-> compile
+- $ exists app/controllers/echo.api.yaml.scala
+$ exists target/scala-2.11/routes/main/controllers_base/echo.api.yaml.scala
+$ exists target/scala-2.11/routes/main/model/echo.api.yaml.scala
+$ exists target/scala-2.11/routes/main/validators/echo.api.yaml.scala
+
+# The controller generation must be triggered explicitly
+> swaggerCompileControllers
+
+# only single controller was generated
 
 $ exists app/controllers/echo.api.yaml.scala
-$ exists test/echo.api.yaml.scala
 $ exists target/scala-2.11/routes/main/controllers_base/echo.api.yaml.scala
-$ exists target/scala-2.11/routes/main/generators/echo.api.yaml.scala
 $ exists target/scala-2.11/routes/main/model/echo.api.yaml.scala
 $ exists target/scala-2.11/routes/main/validators/echo.api.yaml.scala
+- $ exists target/scala-2.11/routes/test/tests/echo.api.yaml.scala
+- $ exists target/scala-2.11/routes/test/generators/echo.api.yaml.scala
+
+# Successfully compiles after controllers have being generated
+> clean
+
+> compile
 
-# we don't check what is created by the play's routes compiler because it can change
+# no test data exists yet
+$ exists app/controllers/echo.api.yaml.scala
+$ exists target/scala-2.11/routes/main/model/echo.api.yaml.scala
+$ exists target/scala-2.11/routes/main/validators/echo.api.yaml.scala
+$ exists target/scala-2.11/routes/main/controllers_base/echo.api.yaml.scala
+- $ exists target/scala-2.11/routes/test/tests/echo.api.yaml.scala
+- $ exists target/scala-2.11/routes/test/generators/echo.api.yaml.scala
 
 # Tests should fail with 404 because there is no mapping in routes file
 -> test
+
+# now everything must be present
+$ exists target/scala-2.11/routes/test/tests/echo.api.yaml.scala
+$ exists target/scala-2.11/routes/test/generators/echo.api.yaml.scala
+$ exists app/controllers/echo.api.yaml.scala
+$ exists target/scala-2.11/routes/main/controllers_base/echo.api.yaml.scala
+$ exists target/scala-2.11/routes/main/model/echo.api.yaml.scala
+$ exists target/scala-2.11/routes/main/validators/echo.api.yaml.scala
+
+# we don't check what is created by the play's routes compiler because its outside of our control
+
```

Comment:
This test script looks like a great foundation.  My approach with scripted tests is to start with something like this, and then as bugs are found and the project evolves, add new ones.  As you would be aware, scripted tests can be a little hard to maintain and slow to run, so it's important that every test delivers a lot of value - I think you'll find this test itself will do just that, and as the project evolves you'll find other areas that would benefit from more testing and so warrant the cost of creating new scripted tests for them.